src/main/java/com/example/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -45,9 +48,15 @@ public void deleteById(Long id) {
     }
 
     @Override
-    public BookDto update(Long id, CreateBookRequestDto requestDto) {
-        Book book = bookMapper.toModel(requestDto);
-        book.setId(id);
-        return bookMapper.toDto(bookRepository.save(book));
+    public BookDto updateById(Long id, CreateBookRequestDto requestDto) {
+        try {
+            Book book = bookMapper.toModel(requestDto);
+            book.setId(id);
+            return bookMapper.toDto(bookRepository.save(book));
+        } catch (EntityNotFoundException e) {
+            throw new EntityNotFoundException(
+                    "The book with ID: id " + id + ", unfortunately was not be founded."
+            );
```

Comment:
I doubt, if we must use that, and think that it's redundant too, and i just wonder, what will be, if input id will be 100000, and we have only 100 books in the db, it turns out that we will have a disorder in the db...

Reply:
Hey @RostyslavKuzmych, thanks for the comment, but I can’t get the main idea of it. 
We have a method that updates our book by ID, also there is a check that validates our ID. 
Question is, what’s the redundant thing?

Reply:
There should be no disorder in the database, but this is provided by another functionality.
Regarding this method, here you need to check if there is a book with the received id, and only if it is - update it, otherwise, throw an error. Rewrite.